### DISCUSSION TITLE
Open API / Swagger Spec Generation

### DISCUSSION THREAD

**Discussion by VinayaSathyanarayana** [prisma/prisma#5757] [Q&A] [answered]
Is there a way to generate Open API / Swagger Spec from Prisma Schema?

**Comment by ryands17** [comment, marked as answer]
@VinayaSathyanarayana 👋 

Currently there is no way to generate an Open API / Swagger spec from a Prisma schema , but you can have a look at [this](https://github.com/pantharshit00/prisma-docs-generator) project that you can take inspiration from to create a generator that creates an Open API spec :)

**Reply by alan345** [reply]
What about this lib? Is it going to do the job? 
https://www.npmjs.com/package/prisma-json-schema-generator

**Reply by Dajcok** [reply]
prisma-json-schema-generator with json-schema-to-openapi-schema can do the job. I had to generate swagger schemas from my prisma models and if you are not using tsao or smth like that these two tools can do the job... After you generate your json you can do: 
```
const fs = require('fs');
const convertSchema = require('json-schema-to-openapi-schema');
const YAML = require('js-yaml');
const path = require('path');

const jsonSchema = require('./schemas/json-schema.json');

const openApiSchema = convertSchema(jsonSchema);

const swaggerDocPath = path.join(__dirname, '../swagger.yaml');
const swaggerDoc = YAML.load(fs.readFileSync(swaggerDocPath, 'utf8'));

swaggerDoc.components = swaggerDoc.components || {};
swaggerDoc.components.schemas = { ...swaggerDoc.components.schemas, ...openApiSchema.definitions };

const finalSwaggerPath = path.join(__dirname, '../swagger.yaml');
fs.writeFileSync(finalSwaggerPath, YAML.dump(swaggerDoc));
```
Your question is more than a year old so if you figured smth out that can work better than what I did I would be more than grateful if you share...

**Comment by daaain** [comment]
You can generate the spec with Zenstack and its OpenAPI plugin: https://zenstack.dev/docs/reference/plugins/openapi

Since Zenstack's schema is a superset of Prisma, you should be able to use it as is.

**Reply by igorkosta** [reply]
Hi @daaain! How would one generate the OpenAPI yaml with this plugin?

**Reply by daaain** [reply]
@igorkosta you need to add a section in the schema like:

```zmodel
plugin openapi {
  provider = '@zenstackhq/openapi'
  prefix = '/api/model/exposed'
  title = 'Prisma model internal API'
  version = '0.1.0'
  description = 'Exposed Prisma DB model API'
  output = 'prisma-model-api.json'
}
```

And then every time you generate the client from schema, the OpenAPI spec will be updated too.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by nitzano** [comment]
I've needed that ability as well so created a new geneator: `prisma-openapi` which can generate `openapi.yaml` from prisma schemas

Have a look here: https://www.npmjs.com/package/prisma-openapi

**Reply by hboisgibault** [reply]
Thanks, this is much appreciated :)

It is working well !
